src/Internal/Payment/State/InitialState.php

```diff
@@ -45,10 +81,32 @@ public function __construct(
 	 * @throws Order_Not_Found_Exception Order could not be found.
 	 */
 	public function process() {
-		$order_id       = $this->get_context()->get_order_id();
-		$payment_method = $this->get_context()->get_payment_method();
-		$this->order_service->set_payment_method_id( $order_id, $payment_method->get_id() );
+		$context  = $this->get_context();
+		$order_id = $context->get_order_id();
+
+		// Start by setting up all local objects.
+		$this->order_service->import_order_data_to_payment_context( $order_id, $context );
```

Comment:
Do you think that this method needs to be inside payment context since it is related to payment context, and not from the order itself?

Reply:
Right now, we're trying to keep the payment context dependency-free, which is why this is not there.

However, I'm looking for a suitable name for a service, which links orders and payments. Something like `OrderPaymentService`? We'll have many small methods like this that we'll need to put there.

Reply:
`OrderPaymentService` looks nice solution to me. Thank you for providing a context regarding the dependency-free payment context.

Reply:
We've definitely got a mixed approach currently, and need to unify it by populating all the context in the same place. We need to consider whether that should be inside or outside the state, and to me, it seems like we have to do it **within the state**.

Here's why: Taking a look at the `get_or_create_customer_id_from_order` a couple of lines below. It is one that involves external calls, and should not be performed until verification has been performed. Looking at the verification within `Prepared_State` of the previous PoC (https://github.com/Automattic/woocommerce-payments/blob/spike/payment-process-states/includes/payment/state/class-prepared-state.php#L195), it seems like customer data is not required there, so at least for Stripe, we should:

1. Populate the context with request and order data.
2. Verify that everything is alright before proceeding.
3. Populate the context with data, which requires API calls.

`InitialState` seems like the place to do that, and the `process` method might just receive the request as a parameter. WDYT?

Reply:
> In the future, I imagine something loading a previous context from the database, so I'll leave the instantiation external.

Yeah, I think so. An example is the required authentication flow (such as using 3DS cards). We would need to stop the state machine, save PaymentContext somewhere, wait for more input for shoppers, and then continue the state machine by re-loading PaymentContext and adding more information to it from PaymentRequest.

Reply:
@marcinbot done: https://github.com/Automattic/woocommerce-payments/pull/7332/commits/43b9f48fe0fb5e596f9d8db3428648a9bb5a43ed